React-Template-master/src/pages/Register.tsx

```diff
@@ -1,6 +1,135 @@
-import React from 'react';
-import { Container } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { Button, TextField, Link, Box, Grid, Typography, useTheme } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { PasswordField } from '../components/PasswordField';
+import { emailValidation } from '../components/EmailValidation';
+import { NEUTRAL } from '../theme/palette';
 
-export function Register(): React.ReactElement {
-    return <Container>Register</Container>;
+interface SignUpFormValues {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
 }
+
+const checkForEmptyInputs = (data: SignUpFormValues): boolean => {
+    const allValues = Object.values(data);
+    for (const value of allValues) {
+        if (value === '') return true;
+    }
+    return false;
+};
+
+export const Register = (): JSX.Element => {
+    const { t } = useTranslation();
+    const theme = useTheme();
+    const [data, setData] = useState({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: ''
+    });
+
+    const setPassword = (password: string): void => {
+        setData({ ...data, password });
+    };
+
+    const [isValidEmail, setIsValidEmail] = useState(true);
+
+    const handleSubmit = async (event: React.FormEvent): Promise<void> => {
+        event.preventDefault();
+        //TODO: Register the user and add all the data into firestore
+    };
+    console.log(isValidEmail);
```

Comment:
@GoutamShanbhag This is a repeated mistake. I didn't expect this after informing you not to put log statements while pushing the code.

Reply:
sorry, I will make sure this will not happen again.